### PR TITLE
numbers generator by Yevhen Sahach

### PR DESCRIPTION
**Pull request by eugeneSahach**

### CODE REVIEW

File: src/createNumbersGenerator.js [hunk 1]
```diff
@@ -26,7 +26,18 @@
  * @return {number}
  */
 function createNumbersGenerator(numbers, min, max) {
-  // write code here
+  const nums = arguments.length > 1
```
Comment (irenhh) [inline src/createNumbersGenerator.js]: add default values

**Review by irenhh — CHANGES_REQUESTED** [review]

File: src/createNumbersGenerator.js [hunk 2]
```diff
@@ -25,8 +25,17 @@
  *
  * @return {number}
  */
-function createNumbersGenerator(numbers, min, max) {
-  // write code here
+function createNumbersGenerator(numbers = [], min, max) {
+  const nums = arguments.length > 1
+    ? numbers.filter(elem => (elem >= min && elem < max))
```
Comment (irenhh) [inline src/createNumbersGenerator.js]: `elem` -> `num`

Comment (irenhh) [reply src/createNumbersGenerator.js]: you don't need this condition

**Review by irenhh — CHANGES_REQUESTED** [review]

**Comment by yevhen-sahach** [comment]
I removed condition, but does it make sense to iterate input if there no min and max specified? I ask it because in case if our input array has huge amount of elements it will iterate through them and spend time instead of taking "ready to go" array

**Comment by irenhh** [comment]
> I removed condition, but does it make sense to iterate input if there no min and max specified? I ask it because in case if our input array has huge amount of elements it will iterate through all of them and spend time instead of taking "ready to go" array

this makes sense, you can leave your version

**Review by irenhh — APPROVED** [review]

**Pull request closed without merging**